Find_first_missing_positive_integer.rb

```diff
@@ -0,0 +1,35 @@
+=begin 
+This problem was asked by Stripe.
+
+Given an array of integers, find the first missing positive integer in linear time and constant space. 
+In other words, find the lowest positive integer that does not exist in the array. 
+The array can contain duplicates and negative numbers as well.
+
+For example, the input [3, 4, -1, 1] should give 2. The input [1, 2, 0] should give 3.
+
+You can modify the input array in-place.
+
+=end 
+
+def missingpostiveint(array)
+    array = array.sort 
+    expected = array[0]
+    array.each_with_index do |value, index|
+        return expected if value != expected
+        expected = 1 if expected == -1
+        return expected += 1  if value == expected && index == array.length-1

```

Comment:
why do we need to check this one. we can always do expected += 1 at this stage? 

even if we are at the last element, we calculate what the next element should be, but the loop will not run again because we only loop till the last element.

Reply:
That's true, but i wasn't getting the expected, for this case [0,1,2] so i had to do in this way.